canonical-web-and-design/maas-ui
# refactor: subnets fabric summary

## Done

- refactor subnets fabric summary
  - add Definition component for consistent styling (and use on Device details page as well)
- add description field to AddFabric

## Screenshots
![image](https://user-images.githubusercontent.com/7452681/150795772-18706aa0-5d4f-41d7-8d12-4f404d768f35.png)
*Add Fabric with new "Description" field*

![Kapture 2022-01-24 at 14 58 23](https://user-images.githubusercontent.com/7452681/150796346-3665c4c7-5474-4e88-b744-63ba23528501.gif)
*subnets fabric summary with loading states*

## QA

### MAAS deployment

To run this branch you will need access to one of the following MAAS deployments:

- [Karura](/HACKING.md#karura)
- [Bolla](/HACKING.md#bolla)
- [A development MAAS](/HACKING.md#development-deployment)
- [A local snap MAAS](/HACKING.md#snap-deployment) (this will not usually have machines)

### Running the branch

You can run this branch by:

- Serving with [dotrun](/HACKING.md#maas-ui-development-setup)
- [Building in a development MAAS](/HACKING.md#running-maas-ui-from-a-development-maas)

### QA steps

- Go to the new fabric URL, e.g. `/MAAS/r/fabric/12` and verify everything looks correctly.

## Fixes

Fixes: # .

## Launchpad issue

Related Launchpad maas issue in the form `lp#number`.

## Backports

In general, please propose fixes against *master* rather than release branches (e.g. 2.7), unless the fix is only applicable for that specific release. Please apply backport labels to the PR (e.g. "Backport 2.7") for the appropriate releases to target.

Only bug and security fixes should be backported, new features should only land in master.
